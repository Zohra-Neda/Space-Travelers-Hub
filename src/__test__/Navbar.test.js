import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Navbar from '../components/Navbar';
// Use MemoryRouter to wrap the Navbar component to simulate routing behavior
const TestNavbar = () => (
  <MemoryRouter>
    <Navbar />
  </MemoryRouter>
);

describe('Navbar', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<TestNavbar />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
