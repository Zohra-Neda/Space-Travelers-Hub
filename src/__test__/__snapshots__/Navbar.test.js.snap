// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly 1`] = `
<nav
  className=" flex justify-between px-10 items-center border-b py-3"
>
  <div
    className="logo-container flex text-4xl font-semibold align-middle items-center"
  >
    <img
      alt="logo"
      className="me-4"
      height="80px"
      src="planet.png"
      width="80px"
    />
    <h2>
      Space Travelers' Hub
    </h2>
  </div>
  <ul
    className="flex"
  >
    <li
      className="me-2"
    >
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
      >
        Rockets
      </a>
    </li>
    <li
      className="me-2"
    >
      <a
        className=""
        href="/missions"
        onClick={[Function]}
      >
        Missions
      </a>
    </li>
    <div
      className=" vertical-line me-2"
    />
    <li
      className="me-2"
    >
      <a
        className=""
        href="/myprofile"
        onClick={[Function]}
      >
        My Profile
      </a>
    </li>
  </ul>
</nav>
`;
